src/verilog/verilog_interfaces.cpp

```diff
@@ -310,29 +310,18 @@ void verilog_typecheckt::interface_function_or_task_decl(const verilog_declt &de
 
   for(auto &declarator : decl.declarators())
   {
-    if(declarator.id() == ID_symbol)
-    {
-      symbol.base_name = declarator.identifier();
-      symbol.type=type;
-    }
-    else if(declarator.id() == ID_array)
-    {
-      symbol.base_name = declarator.identifier();
-      symbol.type = array_type(declarator, type);
-    }
-    else
-    {
-      throw errort().with_location(declarator.source_location())
-        << "unexpected declaration: " << declarator.id();
-    }
+    DATA_INVARIANT(declarator.id() == ID_declarator, "must have declarator");
+
+    symbol.base_name = declarator.base_name();
 
-    if(symbol.base_name=="")
+    if(symbol.base_name == "")
```

Comment:
Nit pick: I'd use `.empty()`just in case there is a more efficient way in `std::string`to check that than the strcmp that the `== ""`will require.